src/mem/ruby/protocol/GPU_VIPER-TCC.sm

```diff
@@ -363,7 +363,9 @@ machine(MachineType:TCC, "TCC Cache")
             } else {
                 trigger(Event:WrVicBlk, in_msg.addr, cache_entry, tbe);
             }
-        } else if (in_msg.Type == CoherenceRequestType:Atomic) {
+        } else if ((in_msg.Type == CoherenceRequestType:Atomic) ||
```

Comment:
Could we perhaps get rid of the use of `CoherenceRequestType:Atomic` in VIPER files entirely? The Return or NoReturn would complete all cases, right?

Reply:
I thought about that as well, and I agree that it would make more sense to remove. However, I though it might make future developments more problematic because now the CPU protocols that use the CPURequestMsg structure use the Atomic type, while the GPU ones use either AtomicReturn or AtomicReturn. I'm not sure what would be a better option. In my opinion, I feel like the GPU protocols should use their own GPURequest and Response messages that are different from the CPU ones, but this would be a pretty significant refactor, and has a large code duplication.